tests/queries/0_stateless/03203_system_query_metric_log.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+
+CURDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+# shellcheck source=../shell_config.sh
+. "$CURDIR"/../shell_config.sh
+
+readonly query_prefix=$CLICKHOUSE_DATABASE
+
+$CLICKHOUSE_CLIENT --query-id="${query_prefix}_1000" -q "SELECT sleep(2.5) FORMAT Null" &
+$CLICKHOUSE_CLIENT --query-id="${query_prefix}_1234" -q "SELECT sleep(2.5) SETTINGS query_metric_log_interval=1234 FORMAT Null" &
+$CLICKHOUSE_CLIENT --query-id="${query_prefix}_123" -q "SELECT sleep(2.5) SETTINGS query_metric_log_interval=123 FORMAT Null" &
+$CLICKHOUSE_CLIENT --query-id="${query_prefix}_0" -q "SELECT sleep(2.5) SETTINGS query_metric_log_interval=0 FORMAT Null" &
+$CLICKHOUSE_CLIENT --query-id="${query_prefix}_fast" -q "SELECT sleep(0.1) FORMAT Null" &
+
+wait
+
+$CLICKHOUSE_CLIENT -q "SYSTEM FLUSH LOGS"
+
+function check_log()
+{
+    interval=$1
+    # We calculate the diff of each row with its previous row to check whether the intervals at which
+    # data is collected is right. The first row is always skipped because the diff is 0. The same for the
+    # last row, which is skipped because doesn't contain a full interval.
+    $CLICKHOUSE_CLIENT --max_threads=1 -m -q """
+    WITH diff AS (
+        SELECT
+            row_number() OVER () AS row,
+            count() OVER () as total_rows,
+            event_time_microseconds,
+            first_value(event_time_microseconds) OVER (ORDER BY event_time_microseconds ROWS BETWEEN 1 PRECEDING AND 0 FOLLOWING) as prev,
+            dateDiff('ms', prev, event_time_microseconds) AS diff
+        FROM system.query_metric_log
+        WHERE query_id = '${query_prefix}_${interval}'
```

Comment:
Please filter by event_date too, so this query doesn't take forever on old systems that have been running for a while (`event_date >= yesterday()`)